dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/LinkageChecker.java

```diff
@@ -89,39 +78,11 @@ public static LinkageChecker create(
         ClassReferenceGraph.create(symbolReferenceMaps, ImmutableSet.copyOf(entryPoints));
 
     return new LinkageChecker(
-        dumper,
-        classPath,
-        symbolReferenceMaps,
-        classReferenceGraph,
-        ExcludedErrors.create(exclusionFile));
-  }
-
-  public static LinkageChecker create(Bom bom) throws IOException {
-    return create(bom, null);
-  }
-
-  public static LinkageChecker create(Bom bom, Path exclusionFile) throws IOException {
-    // duplicate code from DashboardMain follows. We need to refactor to extract this.
-    ImmutableList<Artifact> managedDependencies = bom.getManagedDependencies();
-
-    ClassPathBuilder classPathBuilder = new ClassPathBuilder();
-    ClassPathResult classPathResult = classPathBuilder.resolve(managedDependencies);
-    ImmutableList<ClassPathEntry> classpath = classPathResult.getClassPath();
-
-    // When checking a BOM, entry point classes are the ones in the artifacts listed in the BOM
-    List<ClassPathEntry> artifactsInBom = classpath.subList(0, managedDependencies.size());
-    ImmutableSet<ClassPathEntry> entryPoints = ImmutableSet.copyOf(artifactsInBom);
-
-    return LinkageChecker.create(classpath, entryPoints, exclusionFile);
+        dumper, classPath, symbolReferenceMaps, classReferenceGraph, excludedErrors);
```

Comment:
This is where I think we're going wrong (not in this PR for the first time though). There should only be two arguments here: the classpath and the excludedErrors. dumper, symbolReferenceMaps, and classReferenceGraph are just different views of the classpath. All the information is in the classpath.

ClasspathResult isn't used at this line, but it shouldn't be separate either.

Reply:
Great point. I just created https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/1349/files , and now I'm feeling LinkageChecker only needs a ClassDumper field.


Regarding ClassPathResult v.s. `List<ClassPathEntry>`, ClassPathResult is a result of a class path resolution and  has List<ClassPathEntry> and dependency paths. Simple list of JAR files (without Maven artifacts) does not fit in ClassPathResult.